tests/test_stack.py

```diff
@@ -41,3 +42,21 @@ def test_items_removed_in_lifo_order(stack):
     assert stack.pop() == 3
     assert stack.pop() == 5
     assert stack.empty()
+
+def test_expected_string_returned(stack):
+    stack.push(5)
+    stack.push(10)
+    stack.push(15)
+
+    assert str(stack) == "[15, 10, 5]"
+
+def test_exception_raised(stack):
+    stack.push(5)
+    stack.push(10)
+    stack.push(15)
+    
+    assert stack.pop() == 15
+    assert stack.pop() == 10
+    assert stack.pop() == 5
+    with pytest.raises(EmptyListError, match='List is empty'):
```

Comment:
The stack shouldn't reveal the fact that it's implemented with a `LinkedList`. It should raise a `StackEmptyException` here.

Reply:
When I attempted to raise the `StackEmptyException`, the test failed because the actual error raised was the `EmptyListError` instead. Is there another way to test that the `StackEmptyException` is raised?

Reply:
The underlying problem was that the way the check for an empty list is written, it will never detect the list being empty, so the error you wanted to raise, won't get raised: https://github.com/Ada-C16/stacks-queues/pull/5/files#r860434467